File: Examples/Filtering/MedianImageFilter.py
#
#  Example on the use of the MedianImageFilter
#

from InsightToolkit import *

from sys import argv


reader = itkImageFileReaderUS2_New()
writer = itkImageFileWriterUS2_New()


filter  = itkMedianImageFilterUS2US2_New()

filter.SetInput( reader.GetOutput() )
writer.SetInput( filter.GetOutput() )

reader.SetFileName( argv[1] )
writer.SetFileName( argv[2] )

sizeRadius = itkSize2()

sizeValues = new_ULArray(2)

ULArray_setitem( sizeValues, 0, eval( argv[3] ) )
ULArray_setitem( sizeValues, 1, eval( argv[3] ) )

print ULArray_getitem( sizeValues, 0, )
print ULArray_getitem( sizeValues, 1, )

sizeRadius.SetSize( sizeValues );

filter.SetRadius( sizeRadius )


writer.Update()

File: Examples/Filtering/MedianImageFilter.java
/**
 *  Example on the use of the MedianImageFilter
 *
 */

import InsightToolkit.*;

public class MedianImageFilter
{
  public static void main( String argv[] )
  {
    System.out.println("MedianImageFilter Example");

    itkImageFileReaderUS2_Pointer reader = itkImageFileReaderUS2.itkImageFileReaderUS2_New();
    itkImageFileWriterUS2_Pointer writer = itkImageFileWriterUS2.itkImageFileWriterUS2_New();

    itkMedianImageFilterUS2US2_Pointer filter = itkMedianImageFilterUS2US2.itkMedianImageFilterUS2US2_New();

    filter.SetInput( reader.GetOutput() );
    writer.SetInput( filter.GetOutput() );

    reader.SetFileName( argv[0] );
    writer.SetFileName( argv[1] );


    int radius = Integer.parseInt( argv[2] );

    itkSize2 sizeRadius = new itkSize2();

    sizeRadius.SetElement( 0, radius );
    sizeRadius.SetElement( 1, radius );

    filter.SetRadius( sizeRadius );

    writer.Update();
  }

}


